supporting-blog-content/context-engineering-mistral-completions/notebook.ipynb

```diff
@@ -0,0 +1,517 @@
+    "                                content = choice[\"delta\"][\"content\"]\n",
+    "                                if content:  # Only yield non-empty content\n",
+    "                                    yield content\n",
+    "\n",
+    "                    except json.JSONDecodeError as json_err:\n",
+    "                        # If JSON parsing fails, log the error but continue\n",
+    "                        print(f\"\\nJSON decode error: {json_err}\")\n",
+    "                        print(f\"Problematic data: {data_content}\")\n",
+    "                        continue\n",
+    "\n",
+    "    except requests.exceptions.RequestException as e:\n",
+    "        yield f\"Error: {str(e)}\"\n",
+    "\n",
+    "\n",
+    "print(\"✅ Streaming function defined!\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e0742f3a",
+   "metadata": {},
+   "source": [
+    "## Testing the Inference Endpoint \n",
+    "\n",
+    "Now let's test our inference endpoint with a simple question. This will demonstrate streaming responses are working well from Elasticsearch."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9f5fbf08",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "user_question = \"What SNES games had a character on a skateboard and axes?\"\n",
+    "\n",
+    "messages = [\n",
+    "    {\n",
+    "        \"role\": \"system\",\n",
+    "        \"content\": \"You are a helpful gaming expert that provides concise answers about video games.\",\n",
+    "    },\n",
+    "    {\"role\": \"user\", \"content\": user_question},\n",
+    "]\n",
+    "\n",
+    "print(f\"User: {user_question}\")\n",
+    "print(\"Assistant: \\n\")\n",
+    "\n",
+    "for chunk in stream_chat_completion(\n",
+    "    ELASTICSEARCH_HOST, INFERENCE_ENDPOINT_NAME, messages\n",
+    "):\n",
+    "    print(chunk, end=\"\", flush=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d92b8371",
+   "metadata": {},
+   "source": [
+    "# Context Engineering with Elasticsearch\n",
+    "\n",
+    "In this section, we'll demonstrate how to:\n",
+    "1. Index documents into Elasticsearch \n",
+    "2. Search for relevant context\n",
+    "3. Use retrieved documents to enhance our chat completions with contextual information\n",
+    "\n",
+    "This approach combines retrieval-augmented generation (RAG) with Mistral's chat capabilities through Elasticsearch."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c3dd003a",
+   "metadata": {},
+   "source": [
+    "## Step 1: Index some documents\n",
+    "\n",
+    "First, let's create an Elasticsearch index to store our documents with both text content and vector embeddings for semantic search."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b87d34d3-e64f-49b7-9403-4179e9cbc0e8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
```

Comment:
Please add code to either remove existing index or state that the index already exists. It currently responds with <Response [400]> which can be confusing.